Quest storage

OctoOsmo opened the pull request:
Отключил core модуль и закомментировал лишние неработающие DAO, как и все наверно
Добавил 
- Модуль quest-storage
- view Storage.xhtml

Всё на primefaces
Что вышло:
![2015-12-30 18 25 54](https://cloud.githubusercontent.com/assets/3993121/12052724/d562b3ba-af22-11e5-86c3-581c53091153.png)

OctoOsmo committed:
core module was excluded from main pom
quest-storage module was added
quest-editor-impl classes was temporary fixed

OctoOsmo committed:
QuestStorage view was added (Storage.xhtml)
Quest class, QuestDifficulty enum and chartView were added temporarily
same thing about dashboard view and index pages

Alexander091 commented on quest-storage/src/main/java/DashBoardView.java:
```diff
@@ -0,0 +1,72 @@
+
+    @PostConstruct
+    public void init() {
+        model = new DefaultDashboardModel();
```
Migrate this to DAO

OctoOsmo committed:
questStorageController class was added, it was missed in prev commit for unknown reason

OctoOsmo committed:
quest storage stable version

OctoOsmo committed:
packaging into ear for quest storage was added

OctoOsmo committed:
quest list size was increased

OctoOsmo committed:
Merge remote-tracking branch 'origin/dev-branch' into quest-storage

Conflicts:
	quest-editor/quest-editor-impl/src/main/java/org/textadventure/questeditor/impl/ActionController.java
	quest-editor/quest-editor-impl/src/main/java/org/textadventure/questeditor/impl/ConditionController.java
	quest-editor/quest-editor-impl/src/main/java/org/textadventure/questeditor/impl/NodeController.java
	quest-editor/quest-editor-impl/src/main/java/org/textadventure/questeditor/impl/QuestController.java
	quest-editor/quest-editor-impl/src/main/java/org/textadventure/questeditor/impl/TransitionController.java
	text-adventure-app/text-adventure-app-ear/pom.xml
	text-adventure-app/text-adventure-app-ear/target/application.xml

bogomolovd committed:
added update.sql, testdata.sql. Quest data pulls from db now

OctoOsmo committed:
quest storage beans was moved to quest storage impl

OctoOsmo committed:
quest storage beans was properly moved to quest storage impl

OctoOsmo committed:
Revert "quest storage beans was moved to quest storage impl"

This reverts commit 0c6821600041b7398b541a0967cebabae78322b5.

OctoOsmo committed:
Merge branch 'quest-storage-fixed' into quest-storage

OctoOsmo committed:
Merge remote-tracking branch 'origin/editor' into quest-storage

OctoOsmo committed:
quest storage was connected to database (hibernate)
project index page was updated
primefaces theme support was added

bogomolovd committed:
graph added in graph tab

bogomolovd committed:
added rest service for fetching quest data via ajax (rest service returns test data now), added QuestBean in order to separate business logic

bogomolovd committed:
Quest nodes loading into graph from db now. Edited sql scripts for include position into Node.

bogomolovd committed:
graph displaying properly now

bogomolovd committed:
info about nodes and transitions now loading in inputs. Created rest controllers for node and transition

OctoOsmo committed:
quests can be deleted from queststorage page (by quest id)
welcome file was changed to index.xhtml
obsolete beans was removed
log4j dependence was added into main pom file

OctoOsmo committed:
quest image was added into hibernate mapping, and into database of course
quests on storage view sorted by id now
log4j was removed
JSF templates was added
quests script, which creates some quests and link to images, was added

OctoOsmo committed:
Merge remote-tracking branch 'origin/editor' into quest-storage

Conflicts:
	dao-manager/dao-manager-db/src/main/resources/update.sql

OctoOsmo committed:
sql scripts were updated

OctoOsmo committed:
quests sql script was updated
image_resource was added into quest entity
unified template was inserted into editor.xhtml
index page was changed

Alexander091 commented:
Откати все изменения из quest-editor. Или каким образов в твоей ветке оказались коммиты Димы?

OctoOsmo committed:
quests sql script was updated
image_resource was replace by image in quests table and hibernate mapping
annotations was added into resource DAO
menu template was moved into templates folder

OctoOsmo committed:
Revert "Merge remote-tracking branch 'origin/editor' into quest-storage"

This reverts commit f04caf5467c9aad1cc21faf5e6d756bb141bfa36, reversing
changes made to 808618a821a5c40d8e4665240e70d017e5bc2a5c.

Conflicts:
	dao-manager/dao-manager-db/src/main/resources/testdata.sql
	quest-editor/quest-editor-webapp/src/main/webapp/quest-editor/editor.xhtml

OctoOsmo committed:
Revert "Merge remote-tracking branch 'origin/editor' into quest-storage"

This reverts commit 9e12889411236c1432267bf39f16f902dc1f2c08, reversing
changes made to 4782748c0214b3056900c5c4e59ec7c036261fef.

Conflicts:
	dao-manager/dao-manager-db/src/main/resources/testdata.sql
	dao-manager/dao-manager-db/src/main/resources/update.sql

OctoOsmo committed:
QuestDAO implements commonDAO now

OctoOsmo commented:
Откатил изменения из ветки editor

OctoOsmo commented:
Для истории скриншот текущей версии:
![2016-02-09 04 03 14](https://cloud.githubusercontent.com/assets/3993121/12904670/15e9d434-cee2-11e5-8b15-f73e46528130.png)

OctoOsmo committed:
quest delete confirm dialog was added
